app/assets/javascripts/uploadcare/settings.coffee

```diff
@@ -201,6 +203,16 @@ uploadcare.namespace 'settings', (ns) ->
 
     if settings.validators
       settings.validators = settings.validators.slice()
+    
+    if settings.previewUrlProvider and not settings.overridePreviewUrl
+      settings.overridePreviewUrl = (url, info) => 
+        useGetParam = /\?$/.test(settings.previewUrlProvider)
+        location = if useGetParam then "url=#{encodeURIComponent(url)}" else "/#{url}"
```

Comment:
`&url=` is much safe here, will work for all cases:

`http://domain.com/?` + `&url=`
`http://domain.com/?arg=value` + `&url=`
`http://domain.com/?arg=value&` + `&url=`

Reply:
Another case is when a user provides parameter name, like this: `http://domain.com/?arg=`.
It should be handled with another check for `=` at the end of previewUrlProvider.